File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy=prices[0]
        profit=0
        for i in range(1,len(prices)):
            sell=prices[i]
            if buy>sell:
                buy=sell
            else:
                profit=max(profit,(sell-buy))
        return profit
        

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        int buy=prices[0];
        int maxProfit=0;

        foreach(int item in prices){
            if(item<=buy){
                buy=item;
            }
            else{
                maxProfit=Math.Max(maxProfit,item-buy);
            }
        }
        return maxProfit;
    }
}